### DISCUSSION TITLE
Wrong deprecation of dangerouslyGetParent within type definition file in latest v5.x.x

### DISCUSSION THREAD

**Discussion by SaeedZhiany** [react-navigation/react-navigation#10145] [General]
I'm using the latest available v5.x.x of react-navigation dependencies in my project which is written using Typescript. in somewhere of my code I'm using the `dangerouslyGetParent()` function. but I got a deprecation error on compile time. according to this documentation, I should replace it with the `getParent()` function to get rid of the error.

but in v5 `navigation` prop does not contain the `getParent` function. so in runtime, I got the error message below. I've printed the `navigation` prop in the console and I see no `getParent` key within it

![image](https://user-images.githubusercontent.com/13766432/142843390-48ffae37-92c5-4da2-900e-a54d6ca73b48.png)

As the error says, navigation has not any property as `getParent` so it will be undefined, and calling an undefined function causes the error.

```tsx
const stackNavigation = (isTabScreen ? navigation.getParent() : undefined) ?? navigation;
```

please note that the code below works fine.

```tsx
const stackNavigation = (isTabScreen ? navigation.dangerouslyGetParent() : undefined) ?? navigation;
```

so we have two options here

1) revert `dangerouslyGetParent()` deprecation annotation to get rid of the false error message in react-navigation v5.x.x

2) provide `getParent` function for the `navigation` prop in react-navigation v5.x.x

for now, I have to force my tslint to ignore the deprecation and keep using the `dangerouslyGetParent()` function

```tsx
// tslint:disable-next-line:deprecation TODO use navigation.getParent after upgrading to v6.x.x
const stackNavigation = (isTabScreen ? navigation.dangerouslyGetParent() : undefined) ?? navigation;
```

cc @satya164

**Comment by satya164** [comment]
It does contain `getParent`: https://github.com/react-navigation/react-navigation/blob/6a64c0bae2772bf4e49d3f6d55a2ace9ff84dcee/packages/core/src/useNavigationHelpers.tsx#L87

Don't cc me. I get enough notifications every day.

**Reply by SaeedZhiany** [reply]
It just exists in the definition file, not in runtime, and within the actual `navigation` object. you can see yourself!

here is the navigation object content. it seems the function has not been passed in v5

<img width="643" alt="image" src="https://user-images.githubusercontent.com/13766432/142855809-64cef507-64d9-4e61-a374-7d49b23018fb.png">

**Reply by SaeedZhiany** [reply]
here are versions I'm using in my project. they are the latest versions of react-navigation v5

```
"@react-navigation/drawer": "5.12.9",
"@react-navigation/material-top-tabs": "5.3.19",
"@react-navigation/native": "5.9.8",
"@react-navigation/stack": "5.14.9",
```

**Reply by satya164** [reply]
I linked to source code, not definition file.

**Reply by SaeedZhiany** [reply]
Yeah, I see, and I provide the actual runtime navigation object for you. so obviously there's a bug in v5. also I should mention I create my screens as class components and I get the `navigation` object from prop not using hook functions.

**Reply by satya164** [reply]
> Yeah, I see, and I provide the actual runtime navigation object for you. so obviously there's a bug in v5

It could be anything, stale cache, an issue with dependencies in your project, an issue with a specific code snippet, etc, or a bug in React Navigation. Impossible to say from just a screenshot. You need to reproduce this issue on a snack or minimal repo and open an issue instead if it's a bug.

**Reply by SaeedZhiany** [reply]
are you sure, you have included [the related commit](https://github.com/react-navigation/react-navigation/commit/bc9c046bd29e5272bced8f77348918653283d22e) in a published npm version? I can't find a related changelog about that in [the changelog file](https://github.com/react-navigation/react-navigation/blob/main/packages/native/CHANGELOG.md)

**Reply by satya164** [reply]
> are you sure, you have included the related commit in a published npm version

you wouldn't have the type definitions if it wasn't published. the changelog file you linked is from main branch, not 5.x branch. regardless, it's much faster to have a repro than speculate things
